src/main/java/hudson/security/AuthorizationMatrixProperty.java

```diff
@@ -88,6 +89,13 @@
     private AuthorizationMatrixProperty() {
     }
 
+    @DataBoundConstructor
+    public AuthorizationMatrixProperty(List<String> permissions) {
```

Comment:
Without a matching getter, this does not work for **Snippet Generator**. You would need something like:

```java
@Restricted(DoNotUse.class) // Snippetizer
public List<String> getPermissions() {
    List<String> result = new ArrayList<String>();
    for (Map.Entry<Permission, Set<String>> entry : grantedPermissions.entrySet()) {
        for (String sid : entry.getValue()) {
            result.add(entry.getKey().getId() + ":" + sid);
        }
    }
    return result;
}
```